kube-controllers/pkg/controllers/node/hostendpoints.go

```diff
@@ -17,300 +17,425 @@ package node
-		logrus.WithError(err).Warnf("could not delete host endpoint %q", hepName)
 		return err
 	}
-	c.rl.Forget(rlKey)
 
 	logrus.Infof("deleted hostendpoint %q", hepName)
 	return nil
 }
 
-func (c *autoHostEndpointController) deleteHostendpointWithRetries(ctx context.Context, hepName string) error {
-	for n := 1; n <= 5; n++ {
-		logrus.Debugf("deleting hostendpoint %q. attempt #%v", hepName, n)
-		if err := c.deleteHostendpoint(ctx, hepName); err != nil {
-			switch err.(type) {
-			case errors.ErrorResourceDoesNotExist:
-				logrus.Infof("did not delete hostendpoint %q because it doesn't exist", hepName)
-				return nil
-			default:
-				logrus.WithError(err).Infof("failed to delete host endpoint %q, retrying", hepName)
-				time.Sleep(retrySleepTime)
-				continue
-			}
-		}
-		return nil
-	}
-	return fmt.Errorf("too many retries when deleting hostendpoint %q", hepName)
-}
-
-// isAutoHostendpoint determines if the given hostendpoint is managed by
+// isAutoHostEndpoint determines if the given HostEndpoint is managed by
 // kube-controllers.
-func isAutoHostendpoint(h *api.HostEndpoint) bool {
+func isAutoHostEndpoint(h *api.HostEndpoint) bool {
 	v, ok := h.Labels[hepCreatedLabelKey]
 	return ok && v == hepCreatedLabelValue
 }
 
-// createAutoHostendpoint creates an auto hostendpoint for the specified node.
-func (c *autoHostEndpointController) createAutoHostendpoint(ctx context.Context, n *libapi.Node) (*api.HostEndpoint, error) {
-	hep := c.generateAutoHostendpointFromNode(n)
-	rlKey := rateLimiterItemKey{Type: RateLimitCalicoCreate, Name: hep.Name}
-
-	time.Sleep(c.rl.When(rlKey))
-	res, err := c.client.HostEndpoints().Create(ctx, hep, options.SetOptions{})
+// createAutoHostEndpoint creates the supplied HostEndpoint
+func (c *autoHostEndpointController) createAutoHostEndpoint(hostEndpoint *api.HostEndpoint) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	_, err := c.client.HostEndpoints().Create(ctx, hostEndpoint, options.SetOptions{})
 	if err != nil {
 		logrus.Warnf("could not create hostendpoint for node: %v", err)
-		return nil, err
+		return err
 	}
-	c.rl.Forget(rlKey)
-	return res, nil
+	return nil
+}
+
+// generateDefaultAutoHostEndpointName returns the auto HostEndpoint name for default HostEndpoint
+func (c *autoHostEndpointController) generateDefaultAutoHostEndpointName(nodeName string) string {
+	return fmt.Sprintf("%s-%s", nodeName, hostEndpointNameSuffix)
 }
 
-// generateAutoHostendpointName returns the auto hostendpoint's name.
-func (c *autoHostEndpointController) generateAutoHostendpointName(nodeName string) string {
-	return fmt.Sprintf("%s-auto-hep", nodeName)
+// generateTemplateAutoHostEndpointName returns the name for HostEndpoint created with a template
+func (c *autoHostEndpointController) generateTemplateAutoHostEndpointName(nodeName string, templateName string) string {
+	return fmt.Sprintf("%s-%s-%s", nodeName, templateName, hostEndpointNameSuffix)
```

Comment:
I think it's probably OK, but strictly speaking we could hit name length limits here? What is the max length of a Name for a k8s node and what is the max length for a HostEndpoint?